crate/operator/upgrade.py

```diff
@@ -170,6 +170,7 @@ async def upgrade_cluster(
 
 class UpgradeSubHandler(StateBasedSubHandler):
     @crate.on.error(error_handler=crate.send_update_failed_notification)
+    @crate.timeout(timeout=float(config.CLUSTER_UPDATE_TIMEOUT))
```

Comment:
maybe I'm wrong, but doesn't the runtime for each subhandler already start as soon as they are registered? So this one might be delayed while it is waiting for the `BeforeClusterUpdateSubHandler` with a temporary error [here](https://github.com/crate/crate-operator/blob/master/crate/operator/utils/kopf.py#L137) but for kopf it seems to be running all the time, so the whole process might still time out after 2h already?

Reply:
good point, let me test it